src/input.rs

```diff
@@ -0,0 +1,26 @@
+use bevy::prelude::*;
+use bevy::input::keyboard::KeyboardInput;
+
+use crate::location::Location;
+use crate::direction::Direction;
+use crate::snake;
+
+pub struct Plugin;
+
+impl bevy::prelude::Plugin for Plugin {
+    fn build(&self, app: &mut App) {
+        app.add_system(update_directions);
+    }
+}
+
+fn update_directions(mut query: Query<&mut snake::Moving, (With<Location>, With<snake::Head>)>, mut event_keyboard: EventReader<KeyboardInput>) {
+    for mut moving in query.iter_mut() {
+        for event in event_keyboard.iter() {
```

Comment:
This lets you turn around if you are quick enough!

Reply:
That's why I had the `Moving` component remember the last direction as well...

Reply:
Aha! So what is the desired behaviour here? Should we let the player override their previous choice inside a tick, or one input per tick? Also, I'm thinking ahead to where one snake is played on WASD and another on IJKL and another on arrows...

Reply:
I think the player should be able to override his choice between ticks, but he shouldn't be able to go backwards.